api/tasks.go

```diff
@@ -0,0 +1,58 @@
+package api
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+
+	"github.com/ONSdigital/dp-search-reindex-api/models"
+	"github.com/ONSdigital/dp-search-reindex-api/mongo"
+	"github.com/ONSdigital/log.go/log"
+	"github.com/gorilla/mux"
+)
+
+// CreateTaskHandler returns a function that generates a new TaskName resource containing default values in its fields.
+func (api *JobStoreAPI) CreateTaskHandler(ctx context.Context) http.HandlerFunc {
+	log.Event(ctx, "Creating handler function, which calls CreateTask and returns a new TaskName resource.", log.INFO)
```

Comment:
Fair enough. I must admit that I like to add logging at the beginning of functions, to give context, but I can see that context is not really necessary here since the logs only appear at start up. I've removed them now.